src/conn.rs

```diff
@@ -157,6 +154,12 @@ impl Store {
             sqlite: connection,
         })
     }
+
+    pub fn fast_forward_user_partition(&mut self, new_head: Entid) -> Result<()> {
+        let mut metadata = self.conn.metadata.lock().unwrap();
+        metadata.partition_map.expand_up_to(":db.part/user", new_head);
+        db::update_partition_map(&mut self.sqlite, &metadata.partition_map).map_err(|e| e.into())
```

Comment:
I would rather see the metadata change as a side-effect of updating the DB. The `db::` code is where we'll be able to do compare-and-set, and we already have code to update the metadata after a `transact`.

Reply:
Relatedly: it might be worth thinking about what you're doing here as a special case of renumbering.

Renumbering is the process of taking a space of new remote identifiers, moving any conflicting local identifiers out of the way, and making space locally for the remote identifiers. If there are no conflicting local identifiers, then the second step involves no work.

This is a nice unification because it reduces the number of different code paths…